### ISSUE TITLE
keyboard cannot input if switch user

### ISSUE THREAD

**Issue by JimBrv** [tekezo/Karabiner-Elements#548]
keyboard was disabled while switching user, any input doesn't work and the lock screen will hang forever until force power off and restart.
The latest version kara

macos Sierra 10.12.2
MacBook Pro 13, Mid 2012

Thanks

**Comment by theg1nger** [comment]
Same thing for me! I have to hookup an external USB keyboard to login. As soon as I type the password and hit enter the account logs in and the built in keyboard works again. 

Macbook Pro 13, Mid 2012 
OSX 10.12.2

**Comment by shashurup** [comment]
The same here. I've got a workaround. Switch to the Guest user - it doesn't require a password, and then switch back, without logging the guest user out.

**Comment by spinlock** [comment]
Hi @tekezo , I'm sure it's a  bug. Karabiner-Elements will swallow all grabbed input events and no longer work correctly once I logout or the screen saver is activated.

After reading the source code in `device_grabber.hpp`, I think this problem may be caused by [`manipulator_managers_connector_.invalidate_manipulators()`](https://github.com/tekezo/Karabiner-Elements/blob/master/src/core/grabber/include/device_grabber.hpp#L216).
```c++
213  void unset_profile(void) {
214    profile_ = core_configuration::profile(nlohmann::json());
215
216    manipulator_managers_connector_.invalidate_manipulators();
217  }
```

 This function call will invalidate `post_event_to_virtual_devices_manipulator_`, mark it as invalid and then remove it from `post_event_to_virtual_devices_manipulator_manager_`. But we forget to put this manipulator back in the opposite function `set_profile`. 

So I think maybe this problem can be solved by this patch:

```patch
diff --git a/src/core/grabber/include/device_grabber.hpp b/src/core/grabber/include/device_grabber.hpp
index 50a8ef4..c86ddf2 100644
--- a/src/core/grabber/include/device_grabber.hpp
+++ b/src/core/grabber/include/device_grabber.hpp
@@ -217,6 +217,11 @@ public:
       properties.caps_lock_delay_milliseconds = pqrs::karabiner_virtual_hid_device::milliseconds(profile.get_virtual_hid_keyboard().get_caps_lock_delay_milliseconds());
       virtual_hid_device_client_.initialize_virtual_hid_keyboard(properties);
     }
+
+    post_event_to_virtual_devices_manipulator_manager_.invalidate_manipulators();
+
+    post_event_to_virtual_devices_manipulator_ = std::make_shared<manipulator::details::post_event_to_virtual_devices>();
+    post_event_to_virtual_devices_manipulator_manager_.push_back_manipulator(std::shared_ptr<manipulator::details::base>(post_event_to_virtual_devices_manipulator_));
   }
```

**Comment by tekezo** [comment]
Thanks! 
I've fixed this issue at 0.91.1.
(Fixed by this commit https://github.com/tekezo/Karabiner-Elements/commit/f180051fab459fbf475b4cb59fee7347b6e95733)
